src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.cs

```diff
@@ -601,17 +641,95 @@ private void UpdateDimensionProperties()
 				this.Log().LogDebug($"ScrollViewer setting ViewportHeight={ActualHeight}, ViewportWidth={ActualWidth}");
 			}
 
-			ViewportHeight = this.ActualHeight;
-			ViewportWidth = this.ActualWidth;
+			ViewportHeight = ActualHeight;
+			ViewportWidth = ActualWidth;
 
 			ExtentHeight = (Content as IFrameworkElement)?.ActualHeight ?? 0;
 			ExtentWidth = (Content as IFrameworkElement)?.ActualWidth ?? 0;
 
 			ScrollableHeight = Math.Max(ExtentHeight - ViewportHeight, 0);
 			ScrollableWidth = Math.Max(ExtentWidth - ViewportWidth, 0);
+
+			UpdateComputedVerticalScrollability(invalidate: false);
+			UpdateComputedHorizontalScrollability(invalidate: false);
 		}
 
-		/// <summary>
+		private void UpdateComputedVerticalScrollability(bool invalidate)
+		{
+			var scrollable = ScrollableHeight;
+			var visibility = VerticalScrollBarVisibility;
+			var mode = VerticalScrollMode;
+
+			ComputedVerticalScrollBarVisibility = ComputeScrollBarVisibility(scrollable, visibility);
+			ComputedIsVerticalScrollEnabled = ComputeIsScrollEnabled(scrollable, visibility, mode);
+
+			if (_presenter is null)
+			{
+				return; // Control not ready yet
+			}
+
+			// Support for the native scroll bars (delegated to the native _presenter).
+			_presenter.VerticalScrollBarVisibility = ComputeNativeScrollBarVisibility(visibility, mode, _verticalScrollbar);
+			if (invalidate && _verticalScrollbar is null)
+			{
+				InvalidateMeasure(); // Useless for managed ScrollBar, it will invalidate itself if needed.
+			}
+		}
+
+		private void UpdateComputedHorizontalScrollability(bool invalidate)
+		{
+			var scrollable = ScrollableWidth;
+			var visibility = HorizontalScrollBarVisibility;
+			var mode = HorizontalScrollMode;
+
+			ComputedHorizontalScrollBarVisibility = ComputeScrollBarVisibility(scrollable, visibility);
+			ComputedIsHorizontalScrollEnabled = ComputeIsScrollEnabled(scrollable, visibility, mode);
+
+			if (_presenter is null)
+			{
+				return; // Control not ready yet
+			}
+
+			// Support for the native scroll bars (delegated to the native _presenter).
+			_presenter.HorizontalScrollBarVisibility = ComputeNativeScrollBarVisibility(visibility, mode, _horizontalScrollbar);
+			if (invalidate && _horizontalScrollbar is null)
+			{
+				InvalidateMeasure(); // Useless for managed ScrollBar, it will invalidate itself if needed.
+			}
+		}
+
+		private static Visibility ComputeScrollBarVisibility(double scrollable, ScrollBarVisibility visibility)
+		{
+			// Note: The ScrollMode DOES NOT impact the visibility of the ScrollBar, but just it's hit testability!
+
+			switch (visibility)
+			{
+				case ScrollBarVisibility.Auto when scrollable > 0:
+				case ScrollBarVisibility.Visible:
+					return Visibility.Visible;
+
+				default: // i.e.: Auto when scrollable <= 0; Hidden; Disabled;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5943777)